sklearn/feature_selection/_from_model.py

```diff
@@ -400,6 +436,12 @@ def partial_fit(self, X, y=None, **fit_params):
         if not hasattr(self, "estimator_"):
             self.estimator_ = clone(self.estimator)
         self.estimator_.partial_fit(X, y, **fit_params)
+
+        if hasattr(self.estimator_, "feature_names_in_"):
+            self.feature_names_in_ = self.estimator_.feature_names_in_
+        else:
+            self._check_feature_names(X, reset=True)
```

Comment:
During the first call to `partial_fit`, `feature_names_in_` is set set. In the second, third, etc, calls,  we need to validate the names.

```python
first_call = not hasattr(self, "estimator_")
if first_call:
    self.estimator_ = clone(self.estimator)

self.estimator_.partial_fit(X, y, **fit_params)

if hasattr(self.estimator_, "feature_names_in_"):
    self.feature_names_in_ = self.estimator_.feature_names_in_
else:
    self._check_feature_names(X, reset=first_call)
```

Reply:
Thanks@thomasjpfan. I will adopt this suggestion to the code.